app/services/power_manager/base.rb

```diff
@@ -9,6 +9,21 @@ def self.method_missing(method, *args)
       raise ::Foreman::Exception.new(N_("Invalid power state request: %{action}, supported actions are %{supported}"), { :action => action, :supported => SUPPORTED_ACTIONS })
     end
 
+    SUPPORTED_ACTIONS.each do |method|
+      define_method method do # def start
+        action_entry = action_map[method.to_sym] || {}
```

Comment:
These two lines are a bit confusing for me, what do you want to do here? As far as I can tell this:

* Calls action_map to find the power action to call, if it exists. If not, it creates an empty hash (why?). The `entry` part of the name is what confuses me here, I guess `power_action` or `action` just works.
* Then it creates a hash { :action => power_action_symbol } with the power action to call. For some reason, it needs to create the hash to distinguish between calling `default_action` or the method directly/
* Lastly it tries to find `:result` in action_entry? How does `:result` get in there? 

```ruby
power_action = action_map[power_method] # If it's a hash with indifferent access, no need to call .to_sym
if respond_to? power_action
  send(power_action)
else
  default_action(power_method)
end
```

I think this might work for the first part, but I don't understand how `action_entry`, a local variable, gets the `:result` in there.